### ISSUE TITLE
Weather change causes elemental to respawn claimed with 100% health

### ISSUE THREAD

**Issue by xipies** [DarkstarProject/darkstar#3839]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:**
30170304_1

**_Server Version_** (type `@revision` in game) **:**
0ea9e4a

**_Source Branch_** (master/stable) **:**
master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:**
Fight an elemental for an extended period (e.g., skill up archery), wait for weather to change. Specifically I noticed this when weather would change to double weather. Elemental respawns at its original spawn point, still claimed, and with full health again.

**Comment by Hozu** [comment]
This happens when it changes between single and double weather of the same element, correct?

**Comment by TeoTwawki** [comment]
2 situations where nothing should happen to an elemental: 

- A) the weather changed, but the elemental was engaged to a player

- B) weather changed to diff tier of same element (double/single)

Sounds like code that pops elementals isn't checking that they are already up.

**Comment by xipies** [comment]
This also happened from no weather to weather. I had elemental claimed from when weather was up before. I only noticed this happen when the same weather type was in play. I did not happen to notice/test if weather type changed (e.g., water to lightning).

**Comment by Hozu** [comment]
Makes sense. Probably what @TeoTwawki said - it's not checking if the elemental is already spawned when it does the weather change stuff.
